content/departments/engineering/teams/devops/managed/v1.1/mi1-1_delete_process.md

```diff
@@ -0,0 +1,198 @@
+# Deleting a managed instance
+
+<span class="badge badge-note">SOC2/CI-41</span>
+
+## How to request
+
+1. Create a new issue using [this template](https://github.com/sourcegraph/customer/issues/new?assignees=&labels=team%2Fdevops&template=managed-instance-teardown.md&title=)
+2. Notify Cloud in [#cloud] channel, linking to the issue
+
+SLAs for this can be found with our other [managed instance SLAs](../index.md#slas-for-managed-instances).
+
+---
+
+## Deletion guidance for engineering
+
+Deleting a managed instance is initiated by a CE team member by creating an teardown request. There are a few logical steps to follow.
+
+1.  Confirm the request with the CE Team. Data is not able to be recovered after this process is performed.
+
+1.  Destroy the customer's managed instance infrastructure with Terraform.
+1.  Delete GCP Disk Snapshots
+1.  Delete GCP Project for the managed instance.
+1.  Remove audit monitoring from GCP project.
+1.  Update documentation to remove references of the managed instance.
+
+## Confirm teardown request with the CE Team:
+
+Send a message to the #ce team's Slack channel. An example could be:
+
+```
+Hello, @ce! The <customer name> managed instance is scheduled for teardown in 1 hour. If this is in error, please notify me immediately.
+Teardown request issue: <issue link>
+```
+
+## Destroy the managed instance infrastructure
+
+**Important!** Always start with a fresh terminal session to ensure you do not have any conflicting environment variables.
+
+### Setup
+
+Clone or navigate to the `sourcegraph/deploy-sourcegraph-managed` repository
+
+1.  Navigate into the repository
+
+- `cd deploy-sourcegraph-managed`
+
+1.  Pull in the latest changes
+
+- `git checkout main`
+- `git pull`
+
+1.  Create a branch for the teardown
+
+- `git checkout -b $CUSTOMER/destroy-managed-instance`
+
+1.  Setup the environment:
+
+- `export TF_VAR_opsgenie_webhook=<value from 1Password>`
+- `export CUSTOMER=<customer>`
+
+### Navigate to the customer's managed instance directory
+
+```
+cd $CUSTOMER
+```
+
+### Allow the GCP KMS Crypto Key and CloudSQL instance to be deleted
+
+- add extra variables in
+
+```sh
+module "managed_instance" {
+  cloud_sql_deletion_protection  = false
+  source                         = "../modules/terraform-managed-instance-new"
+  (..)
+}
+```
+
+- modify kms key to remove delete protection
+
+```sh
+sed -i '' 's/    prevent_destroy = true/    prevent_destroy = false/g' ../modules/terraform-managed-instance-new/infrastructure.tf
```

Comment:
@daxmc99 should we mention somehow that this change is temporary and CANNOT be committed in PR?

Reply:
Hmm, could we run it within the `.terraform` folder instead? That folder will be removed as part of the deletion.

Reply:
This does not work, as seems like kms is not inside `.terraform`... I added single line to restore module after destroying the instance.